Yank/restraints.py

```diff
@@ -475,6 +552,29 @@ def _add_force_in_separate_group(cls, system, restraint_force):
         restraint_force.setForceGroup(min(available_force_groups))
         system.addForce(restraint_force)
 
+    @property
+    def _excluded_init_parameters(self):
```

Comment:
I wonder if we should create something like an `OrientationalRestraint` base class, and push `_excluded_init_parameters` and `_parameters` into that. The reason being that these two attributes make assumptions about a constructor that `ReceptorLigandRestraint` doesn't have, and, for a general restraint, we want to keep the constructor as free as possible.

If you would like to avoid a base class for `Boresch` and `RMSD`, we could instead have a mix-in class that exposes these two properties so that we can keep the very base restraint class as general as possible.